File: leetcode/two_pointers/dart/container_with_most_water.dart
import 'dart:math';

class Solution {
  int maxArea(List<int> height) {
    int left = 0, right = height.length - 1;
    int result = 0;
    while (left < right) {
      int area = (right - left) * min(height[left], height[right]);
      result = max(result, area);
      if (height[left] < height[right]) {
        left++;
      } else {
        right--;
      }
    }
    return result;
  }
}

void main(List<String> args) {
  final height = [1, 8, 6, 2, 5, 4, 8, 3, 7];
  final object = Solution();
  print(object.maxArea(height));
}

File: leetcode/two_pointers/go/container_with_most_water.go
package two_pointers

import "fmt"

func maxArea(height []int) int {
	result := 0
	left := 0
	right := len(height) - 1
	for left < right {
		area := (right - left) * min(height[left], height[right])
		result = max(result, area)
		if height[left] < height[right] {
			left++
		} else {
			right--
		}
	}
	return result
}

func main() {
	height := []int{1, 8, 6, 2, 5, 4, 8, 3, 7}
	fmt.Println(maxArea(height))
}
